Rivet/Functions/Convert-FileInfoToMigration.ps1

```diff
@@ -81,13 +73,29 @@ function Convert-FileInfoToMigration
             }
         }
 
+        function Clear-Migration
+        {
+            ('function:Push-Migration','function:Pop-Migration') |
+                Where-Object { Test-Path -Path $_ } |
+                Remove-Item -WhatIf:$false -Confirm:$false
+        }
+
+        Clear-Migration
+
+        Import-RivetPlugin -Path $Session.PluginPaths -ModuleName $Session.PluginModules
+    }
+
+    process
+    {
         foreach( $fileInfo in $InputObject )
         {
-            $dbName = Split-Path -Parent -Path $fileInfo.FullName
-            $dbName = Split-Path -Parent -Path $dbName
-            $dbName = Split-Path -Leaf -Path $dbName
+            $dbName = $fileInfo.DatabaseName
+
+            Connect-Database -Session $Session -Name $dbName

```

Comment:
Should this function call `Disconnect-Database` once it's done with its work?

Reply:
Technically, yes. But since it isn't making any database changes, it doesn't need to.

Really, it shouldn't be calling `Connect-Database` at all, but it has to because `Convert-FileInfoToMigration` is called from two different code paths: `Update-Database` (used to push/pop migrations) and `Get-Migration` (returns migration objects).  The `Get-Migration` function doesn't manage the database connection, so `Convert-FileInfoToMigration` has to.

Thinking about the answer to your question I've realized the only public function Rivet should have that isn't an operation or column creation function is `Invoke-Rivet`/`rivet`. This should make it easier to ensure property connection management.